src/utils/serdes/serdes.cpp

```diff
@@ -14,121 +14,156 @@
 
-    ssize_t len;
-
-    //get len
-    //_stringToBytes(&len, workingStr.data() + des_offset + tag.size(), sizeof(ssize_t));
-    _stringToBytes(&len, workingStr.substr(des_offset + tag.size(), sizeof(ssize_t)), sizeof(ssize_t));
+    const std::size_t len = peekLenUnsafe();
+    offset_ += sizeof(len);
 
-    return len;
+    if(offset_ + len + 1 > buffer_.size()) {
+        // throw std::runtime_error("deserialization data insufficient");
+        return "";
+    }
+    const std::string ret = buffer_.substr(offset_, len);
+    offset_ += len + 1;  // Also skip trailing '|'.
+    return ret;
 }
 
-nixl_status_t nixlSerDes::getBuf(const std::string &tag, void *buf, ssize_t len){
-    if(workingStr.compare(des_offset, tag.size(), tag) != 0){
-       //incorrect tag
-       return NIXL_ERR_MISMATCH;
+std::size_t nixlDeserializer::peekBufLen(const std::string_view& tag) const
+{
+    if(offset_ + tag.size() + sizeof(std::size_t) > buffer_.size()) {
+        // throw std::runtime_error("deserialization data insufficient");
+        return -1;
     }
+    if(buffer_.compare(offset_, tag.size(), tag) != 0) {
+        // throw std::runtime_error("deserialization tag mismatch ...");
+        return -1;
+    }
+    return peekLenUnsafe(tag.size());
+}
 
-    //skip over tag and size, which we assume has been read previously
-    des_offset += tag.size() + sizeof(ssize_t);
-
-    //_stringToBytes(buf, workingStr.data() + des_offset, len);
-    _stringToBytes(buf, workingStr.substr(des_offset, len), len);
-
-    //bytes in string form are twice as long, skip those plus | delimiter
-    des_offset += len + 1;
+nixl_status_t nixlDeserializer::getBuf(const std::string_view& tag, void *buf, const std::size_t len)
+{
+    if(offset_ + tag.size() + sizeof(std::size_t) > buffer_.size()) {
+        // throw std::runtime_error("deserialization data insufficient");
+        return NIXL_ERR_MISMATCH;
+    }
+    if(buffer_.compare(offset_, tag.size(), tag) != 0) {
+        // throw std::runtime_error("deserialization tag mismatch ...");
+        return NIXL_ERR_MISMATCH;
+    }
+    if(peekBufLen(tag) != len) {  // NIXL_ASSERT(peekBufLen(tag) == len);
+        // throw std::runtime_error("deserialization size mismatch");
+        return NIXL_ERR_MISMATCH;
+    }
+    offset_ += tag.size() + sizeof(std::size_t);
 
+    if(offset_ + len + 1 > buffer_.size()) {
+        // throw std::runtime_error("deserialization data insufficient");
+        return NIXL_ERR_MISMATCH;
+    }
+    std::memcpy(buf, buffer_.data() + offset_, len);
+    offset_ += len + 1;  // Also skip trailing '|'.
     return NIXL_SUCCESS;
 }
 
-// Buffer management serialization
-std::string nixlSerDes::exportStr() const {
-    return workingStr;
-}
-
-nixl_status_t nixlSerDes::importStr(const std::string &sdbuf) {
-
-    if(sdbuf.compare(0, 11, "nixlSerDes|") != 0){
-       //incorrect tag
-       return NIXL_ERR_MISMATCH;
+nixl_status_t nixlDeserializer::importStr(std::string raw)
```

Comment:
Is there a real use-case for this method? On the cursory glance it seems that all users can just use the constructor directly.